### PR TITLE
Hows My Ssl test for Android

### PR DESCRIPTION
**Pull request by yschimke**
With session ticket disabled in AndroidSocketAdapter

```
09-07 12:04:35.577 24546 25651 W OkHttp  : results HowsMySslResults(unknown_cipher_suite_supported=false, beast_vuln=false, session_ticket_supported=false, tls_compression_supported=false, ephemeral_keys_supported=true, rating=Probably Okay, tls_version=TLS 1.3, able_to_detect_n_minus_one_splitting=false, insecure_cipher_suites={}, given_cipher_suites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_256_GCM_SHA384, TLS_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_256_CBC_SHA])
```

### CODE REVIEW

**Comment by yschimke** [comment]
n.b. above some rate, expect failures unless Square subscribe https://subscriptions.howsmyssl.com/

So we may not want to land, but using for testing in a console.

File: android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt [hunk 1]
```diff
@@ -137,6 +144,42 @@ class OkHttpTest {
...
+    }
+
+    Platform.get().log(Platform.WARN, "results $results", null)
+
+    assertTrue(results.session_ticket_supported)
+    assertEquals("Probably Okay", results.rating)
+    assertEquals("TLS 1.3", results.tls_version)
+    assertEquals(0, results.insecure_cipher_suites.size)
+
+    assertEquals(TlsVersion.TLS_1_3, response.handshake?.tlsVersion)
```
Comment (yschimke) [inline android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt]: TODO adapt this and other checks based on Android version.

File: android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt [hunk 2]
```diff
@@ -137,6 +144,42 @@ class OkHttpTest {
...
+    val tls_compression_supported: Boolean,
+    val ephemeral_keys_supported: Boolean,
+    val rating: String,
+    val tls_version: String,
+    val able_to_detect_n_minus_one_splitting: Boolean,
+    val insecure_cipher_suites: Map<String, List<String>>,
+    val given_cipher_suites: List<String>?
+  )
+
+  @Test
```
Comment (swankjesse) [inline android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt]: Wanna `@Ignore` this so running our test suite by default doesn’t flake on connectivity?

Comment (swankjesse) [reply android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt]: ... I’d love to share tests like this between Android and JVM

File: android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt (hunk 1 above)
Comment (swankjesse) [reply android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt]: Yeah, this is a drag I guess we just need an expected TLS version for each Android version we support.

File: android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt (hunk 2 above)
Comment (yschimke) [reply android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt]: Yep, I'll follow up on that.

**Pull request merged**